scripts/lib-franklin.js

```diff
@@ -431,6 +432,53 @@ export function buildBlock(blockName, content) {
   return (blockEl);
 }
 
+/**
+ * Executes a function with the given context and arguments.
+ * An arrow function will have the context as last parameter, while a regular function will also
+ * have the `this` variable set to the same context.
+ * @param {Function} fn the function to execute
+ * @param {Object[]} args the function arugments
+ * @param {Object} context the execution context to use
+ * @returns the result of the function execution
+ */
+function runFunctionWithContext(fn, args, context) {
+  return fn.toString().startsWith('function')
+    ? fn.call(context, ...args, context)
```

Comment:
is it worth to use `.call`? 
It feels a bit strange to have methods which are defined in `aem-lib` (or `lib-franklin`) in the `this` object of a plugin function.

Reply:
The idea behind it was:
- simplify imports and given immediate access to all "core" functions
- allow for "overriding" core functions by modifying the context if needed, as opposed to imports which are read-only (think icons decoration via svg spriting for instance that could replace the regular `decorateIcons`)
- simplify unit testing, since you don't have to inject mocked dependencies anymore